web/app/routes/mainRoutes.js

```diff
@@ -0,0 +1,83 @@
+var express = require("express");
+var router = express.Router();
+var mongoose = require("mongoose");
+mongoose.connect("mongodb://localhost/routerdb");
+
+var userCtrl = require("../controllers/userCtrl");
+var drivingSchoolCtrl = require("../controllers/drivingSchoolCtrl");
+var drivingSessionCtrl = require("../controllers/drivingSessionCtrl");
+
+// server routes ===========================================================
+// handle things like api calls
+// authentication routes
+
+// Get the JSON for the student with the specified _id
+router.get("/students/:_id", function(req, res) {
+  userCtrl.getStudent(req, res);
```

Comment:
Not gonna lie, not a big fan of the whole request getting just passed a layer deeper for no real reason.
I think each of the gets/posts should handle direct req/res info and only pass the information as needed.
E.g.
```
router.get("/students/:_id", function(req, res) {
    userCtrl.getStudent(req.body, (student) => {
        res.statusCode = 201;
        res.json(student);
    }, (errMsg) => {
        res.statusCode = 400;
        res.send(errMsg);
    });
}
```

and
```
exports.createStudent = function(userInfo, callback, error) {
    var newStudent = new User({
        loginDetails: {
            userId: userInfo.loginDetails.userId,
            service: userInfo.loginDetails.service
        },
        firstName: userInfo.body.firstName,
        lastName: userInfo.body.lastName,
        userType: "student"
   });

    if(!userInfo.firstName){
        error("Missing first name");
    } else if(!userInfo.lastName) {
        error("Missing last name");
    } else {
        newStudent.save(function(err) {
            if(!err) {
                callback(newStudent);
            } else {
                error(`Error creating a new student, ${err}`);
            }
        });
    }
};
```
This will actually make the code much easier to test also.